dashboard/app/jobs.go

```diff
@@ -1073,16 +1073,16 @@ func doneJob(c context.Context, req *dashapi.JobDoneReq) error {
 				log.Errorf(c, "job %v: duplicate build %v", jobID, req.Build.ID)
 			}
 		}
-		if job.Log, err = putText(c, ns, textLog, req.Log, false); err != nil {
+		if job.Log, err = putText(c, ns, textLog, req.Log); err != nil {
 			return err
 		}
-		if job.Error, err = putText(c, ns, textError, req.Error, false); err != nil {
+		if job.Error, err = putText(c, ns, textError, req.Error); err != nil {
 			return err
 		}
-		if job.CrashLog, err = putText(c, ns, textCrashLog, req.CrashLog, false); err != nil {
+		if job.CrashLog, err = putText(c, ns, textCrashLog, req.CrashLog, cutInMiddle); err != nil {
```

Comment:
Should the options be attached to text type?
Would it ever be not a copy-paste bug to save textCrashLog w/o cutInMiddle, or save textKernelConfig w/o dedupText, or save textCrashLog with dedupText?
Maybe it's better to have a map of type -> options? Do we ever use the same type with different options?

Reply:
Switched to the maps-based approach.